src/engine/Operation.cpp

```diff
@@ -159,8 +159,9 @@ std::shared_ptr<const Result> Operation::getResult(
         result.applyLimitOffset(_limit);
         runtimeInfo().addLimitOffsetRow(_limit, limitTimer.msecs(), true);
       } else {
-        AD_CONTRACT_CHECK(result.idTable().numRows() ==
-                          _limit.actualSize(result.idTable().numRows()));
+        auto numRows = result.idTable().numRows();
+        AD_CONTRACT_CHECK(numRows ==
+                          std::min(_limit._limit.value_or(numRows), numRows));
```

Comment:
`_limit.acrualSize(numRows)` does exactly that

Reply:
It does not if `offset != 0` and actually leads to false positive assertion failures in those cases. That's why I had to change it.

Reply:
I think we should talk about this. In this case it seems like `_limit.upperBound(numRows)` might be what you're searching for, but it conveys a different meaning and might cause issues with subqueries with offset clauses